app/lib/employment_income_subtotals.rb

```diff
@@ -53,10 +53,18 @@ def entitles_child_care_allowance?
     employments_excluding_self_employments.any?(&:entitles_childcare_allowance?)
   end
 
+  def employment_names
+    [@employment_result].compact.flat_map(&:employments).map(&:employment_name)
+  end
+
```

Comment:
This would be better as 1 method and then let the caller call employment_name and employment_payments methods, thus avoiding the zip call